lib/errorlogger.cpp

```diff
@@ -1022,9 +1038,13 @@ std::string getGuideline(const std::string &errId, ReportType reportType,
                            guideline.begin(), static_cast<int (*)(int)>(std::toupper));
         }
         break;
-    case ReportType::misraC:
+    case ReportType::misraC2012:
+    case ReportType::misraC2023:
+    case ReportType::misraC2025:
         if (errId.rfind("misra-c20", 0) == 0 || errId.rfind("premium-misra-c-20", 0) == 0)
             guideline = errId.substr(errId.rfind('-') + 1);
+        if (errId.find("-dir-") != std::string::npos)
+            guideline = "Dir " + guideline;
```

Comment:
This can still be executed even if the error is not a misra c error. misra c++ also has directives and users can enable both misra c and misra c++ at the same time. And then the guideline will be just "Dir " and no number.